File: HomeExpenses.Server/Models/Category.cs
using System.Text.Json.Serialization;

namespace HomeExpenses.Server.Models
{
    public class Category
    {
        public int Id { get; set; }
        public string Description { get; set; } = string.Empty;
        public CategoryType CategoryType { get; set; }
    }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum CategoryType
    {
        Revenue,
        Expense,
        Both
    }
}

File: homeexpenses.client/src/models/category.ts
export const CategoryTypeOptions = [
    { value: 'Revenue', label: 'RECEITA' },
    { value: 'Expense', label: 'DESPESA' },
    { value: 'Both', label: 'AMBAS' },
] as const;

export type CategoryType =
    typeof CategoryTypeOptions[number]['value'];

export interface Category {
    id: number;
    description: string;
    categoryType: CategoryType;
}

